src/SqlBindingUtilities.cs

```diff
@@ -290,5 +291,67 @@ internal static bool IsFatalSqlException(this Exception e)
             }
             return true;
         }
+
+        /// <summary>
+        /// Query the ServerProperty('EngineEdition') for logging the Server Type for Telemetry
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Throw if an error occurs while querying the engine edition</exception>
+        public static async Task<ServerProperties> GetSqlServerEditions(SqlConnection connection, ILogger logger, CancellationToken cancellationToken)
+        {
+            if (TelemetryInstance.Enabled)
+            {
+                string selectServerEdition = $"SELECT SERVERPROPERTY('EngineEdition'), SERVERPROPERTY('EngineEdition')";
+
+                logger.LogDebugWithThreadId($"BEGIN GetSqlServerEdition Query={selectServerEdition}");
+                using (var selectServerEditionCommand = new SqlCommand(selectServerEdition, connection))
+                using (SqlDataReader reader = await selectServerEditionCommand.ExecuteReaderAsync(cancellationToken))
+                {
+                    if (!await reader.ReadAsync(cancellationToken))
+                    {
+                        throw new InvalidOperationException($"Received empty response when querying the engine edition.");
```

Comment:
I don't think we should throw in this case - since it's only for telemetry we want to continue on even if an error occurs (it just won't include these values in the telemetry which is fine enough)